cpp/cppsdk/gsdk.cpp

```diff
@@ -466,12 +470,12 @@ namespace Microsoft
                 return 0;
             }
 
-            const std::string& GSDK::getLogsDirectory()
+            const std::string GSDK::getLogsDirectory()
             {
                 // Declare as static so that it doesn't live on the stack (since we're returning a reference)
                 static const std::string empty = "";
 
-                const std::unordered_map<std::string, std::string>& config = GSDK::getConfigSettings();
```

Comment:
To avoid an extra copy, we should just lock in this method and access m_configSettings directly.